keps/sig-node/5018-dra-adminaccess/README.md

```diff
@@ -0,0 +1,687 @@
+
+1. Only users with access to the admin namespace can reference these
+   `ResourceClaims` or `ResourceClaimTemplates` in their pod or deployment
+   specs.
+
+1. The DRA controller processes the request and grants privileged access to the
+   requested device.
+
+## Design Details
+
+### API Changes
+
+The `adminAccess` field in `DeviceRequest` was added to `ResourceClaim` and
+`ResourceClaimTemplate` in Kubernetes v1.31:
+
+```go
+// DeviceRequest is a request for devices required for a claim.
+// This is typically a request for a single resource like a device, but can
+// also ask for several identical devices.
+//
+// A DeviceClassName is currently required. Clients must check that it is
+// indeed set. It's absence indicates that something changed in a way that
+// is not supported by the client yet, in which case it must refuse to
+// handle the request.
+type DeviceRequest struct {
+	...
+
+    // AdminAccess indicates that this is a claim for administrative access
+    // to the device(s). Claims with AdminAccess are expected to be used for
+    // monitoring or other management services for a device.  They ignore
+    // all ordinary claims to the device with respect to access modes and
+    // any resource allocations.
+    //
+    // This is an alpha field and requires enabling the DRAAdminAccess
+    // feature gate. Admin access is disabled if this field is unset or
+    // set to false, otherwise it is enabled.
+    //
+    // +optional
+    // +featureGate=DRAAdminAccess
+    AdminAccess *bool
+}
+```
+
+Admin access to devices is a privileged operation because it grants users access
+to devices that are in use by other users. Drivers might also remove other
+restrictions when preparing the device.
+
+In Kubernetes 1.31 (before this KEP was introduced), an example validating
+admission policy
+[was provided](https://github.com/kubernetes/kubernetes/blob/4aeaf1e99e82da8334c0d6dddd848a194cd44b4f/test/e2e/dra/test-driver/deploy/example/admin-access-policy.yaml#L1-L11)
+which restricted access to this option. It was formerly the responsibility of
+cluster admins to ensure that such a policy was installed if the cluster
+shouldn't have allowed unrestricted access.
+
+Starting in Kubernetes 1.33 (when this KEP was introduced), a validation has
+been added to the REST storage layer to only authorize `ResourceClaim` or
+`ResourceClaimTemplate` with `adminAccess: true` requests if their namespace has
+the `kubernetes.io/dra-admin-access` label to only allow it for users with
+additional privileges. More time is needed to figure out how that should work,
+therefore the field is placed behind the `DRAAdminAccess` feature gate.
+
+The `DRAAdminAccess` feature gate controls whether users can set the
+`adminAccess` field to true when requesting devices. That is checked in the
+apiserver. In addition, the scheduler will not allocate claims with admin access
+when the feature is turned off. This covers the scenario that the field might
```

Comment:
Perhaps the scheduler should check for the label as well?

Reply:
eh, letting the consumers just look at the boolean seems ok to me